### PR TITLE
[#ICC-327] Migrate node18

### PR DESCRIPTION
**Pull request by Garma00**
#### List of Changes

* [update node version in .nvmrc and .node-version](https://github.com/pagopa/io-functions-assets/commit/1a385aece3e88429c65d18b1a292d71d118a9020);
* [regenerate yarn.lock](https://github.com/pagopa/io-functions-assets/commit/45caac12dba31559e18fb84a4554215fc06ccbeb);
* [fix type problems](https://github.com/pagopa/io-functions-assets/commit/c2155c0327e0158cf5edd1cf52dafd4ee57fe986);

#### Motivation and Context

End of life Node 14

#### How Has This Been Tested?

Build and unit test

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -544,19 +855,19 @@
...
 "@pagopa/express-azure-functions@^2.0.0":
-  version "2.0.0"
-  resolved "https://registry.yarnpkg.com/@pagopa/express-azure-functions/-/express-azure-functions-2.0.0.tgz#eb52a0b997d931c1509372e2a9bea22a8ca85c17"
-  integrity sha512-IFZqtk0e2sfkMZIxYqPORzxcKRkbIrVJesR6eMLNwzh1rA4bl2uh9ZHk1m55LNq4ZmaxREDu+1JcGlIaZQgKNQ==
+  version "2.1.1"
+  resolved "https://registry.yarnpkg.com/@pagopa/express-azure-functions/-/express-azure-functions-2.1.1.tgz#ac6aa6d8c8d82b64e0db19ccadd2e89cc35ea725"
+  integrity sha512-x2kBGS6JKYqVp7PS9QiBc2AWIOPBObHwlaheJmxZJacnxmLdXHobomTzN2UJMxu55fxMd7xIxqM8VGMQxSzqiw==
 
 "@pagopa/io-functions-commons@^21.0.0":
-  version "21.5.2"
```
Comment (fabriziopapi) [inline yarn.lock:552]: why this version changed? 
there is a not-committed package.json or it changed due to ^ ?

in the former case, could be better to explicitly update the commons version in the package.json?

Comment (Garma00) [reply yarn.lock:552]: Probably something happened during the yarn.lock regeneration... anyway i have updated now `fn-commons` so it is based on Node 18 here [52ea2c7](https://github.com/pagopa/io-functions-assets/pull/77/commits/52ea2c7e0c0ed8f5c5e4a33d4c7553bbec70f9a3)

**Review by fabriziopapi — APPROVED** [review]

File: GetServiceMetadata/__tests__/handler.test.ts [hunk 2]
```diff
@@ -23,15 +23,18 @@ import { ServiceScopeEnum } from "@pagopa/io-functions-commons/dist/generated/de
...
   jest.restoreAllMocks();
 });
 
 const anOrganizationFiscalCode = "01234567890" as OrganizationFiscalCode;
-const aServiceMetadata = {
-  scope: ServiceScopeEnum.NATIONAL
+const aServiceMetadata: ServiceMetadata = {
+  scope: ServiceScopeEnum.NATIONAL,
+  category: StandardServiceCategoryEnum.STANDARD,
+  customSpecialFlow: undefined
```
Comment (michaeldisaro) [inline GetServiceMetadata/__tests__/handler.test.ts:37]: Why are these changes here? Are they correlated with Node18 update or just "boyscout rule" to keep things clean and updated?

File: Info/handler.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ type InfoHandler = () => Promise<
 
 type HealthChecker = (
   config: unknown
-) => healthcheck.HealthCheck<healthcheck.ProblemSource, true>;
+) => healthcheck.HealthCheck<"AzureStorage" | "Config" | "AzureCosmosDB", true>;
```
Comment (michaeldisaro) [inline Info/handler.ts:28]: ditto

**Review by michaeldisaro — APPROVED** [review]

**Review by michaeldisaro — APPROVED** [review]
Just minor comments

File: GetServiceMetadata/__tests__/handler.test.ts (hunk 2 above)
Comment (Garma00) [reply GetServiceMetadata/__tests__/handler.test.ts:37]: After the `yarn.lock` regeneration `tsc` was complaining where this `aServiceMetadata` was used cause it is not a valid `ServiceMetadata` without those 2 fields

File: Info/handler.ts (hunk 3 above)
Comment (Garma00) [reply Info/handler.ts:28]: Yess this is related to the `@pagopa/io-functions-commons` update
